File: core/kora-app-annotation-processor/src/main/java/io/koraframework/kora/app/annotation/processor/interceptor/ComponentInterceptor.java
package io.koraframework.kora.app.annotation.processor.interceptor;

import io.koraframework.kora.app.annotation.processor.component.ResolvedComponent;

import javax.lang.model.type.TypeMirror;

public record ComponentInterceptor(ResolvedComponent component, TypeMirror interceptType) {
}

File: core/kora-app-symbol-processor/src/main/kotlin/io/koraframework/kora/app/ksp/interceptor/ComponentInterceptor.kt
package io.koraframework.kora.app.ksp.interceptor

import com.google.devtools.ksp.symbol.KSType
import io.koraframework.kora.app.ksp.component.ResolvedComponent
import io.koraframework.kora.app.ksp.declaration.ComponentDeclaration

data class ComponentInterceptor(
    val component: ResolvedComponent,
    val declaration: ComponentDeclaration,
    val interceptType: KSType
)
